release/release_tests.yaml

```diff
@@ -47,9 +47,9 @@
 #  # Run configuration for the test
 #  run:
 #    # Type of test. Can be job, sdk_command or client.
-#    # Uses either Ray jobs, anyscale SDK commands or the Ray client to run the actual
-#    # release test.
-#    type: job
+#    # Uses either Ray jobs, anyscale jobs, anyscale SDK commands
+#    # or the Ray client to run the actual release test.
+#    type: anyscale_job
 #
```

Comment:
We should land the changes in this file in a separate PR.

1. Merge anyscale jobs integration with one job running on anyscale_job as a PoC
2. Merge release_tests.yaml config change that converts the rest of the jobs to anyscale_jobs

This is so we can revert 2) without reverting 1) if something goes wrong.